File: daggermvp/src/main/java/be/vergauwen/simon/daggermvp/ui/TestPresenter.java
package be.vergauwen.simon.daggermvp.ui;

import android.content.Context;
import android.util.Log;
import be.vergauwen.simon.himura.MVPPresenter;
import javax.inject.Inject;

public class TestPresenter extends MVPPresenter<TestContract.View>
    implements TestContract.Presenter {

    private Context context;

    @Inject
    public TestPresenter(Context context) {
        this.context = context;
    }

    @Override
    public void attachView(TestContract.View view) {
        super.attachView(view);
    }

    @Override
    public void loadThings() {
        Log.e("TestPresenter", ".loadThings()");
        getView().showThings();
    }
}

File: app/src/main/kotlin/be/vergauwen/simon/ui/TestPresenter.kt
package be.vergauwen.simon.ui

import android.content.Context
import android.util.Log
import be.vergauwen.simon.mvp.MVPPresenter
import javax.inject.Inject

class TestPresenter
@Inject
constructor(private val context: Context) : MVPPresenter<TestContract.View>(), TestContract.Presenter {

  override fun loadThings() {
    Log.e("TestPresenter", ".loadThings()")
    getView()?.showThings()
  }
}
